source/codegen/templates/service_helpers.mako

```diff
@@ -63,8 +63,9 @@ ${set_response_values(output_parameters)}\
 %>\
 ${initialize_input_params(function_name, parameters)}\
 ${initialize_output_params(output_parameters)}\
-% if function_name == config['close_function']:
-      session_repository_->remove_session(${service_helpers.create_args(parameters)});
+% if function_name == config['close_function'] or service_helpers.is_custom_close_method(function_data):
```

Comment:
I suggested an update after seeing this in #47 . Make sure if you make that update you carry it over here. ([Link to comment on other PR](https://github.com/doshirohan/grpc-device/pull/47/files#r624017907))

Reply:
@reckenro I had tried that earlier. But the following system test started to fail:
 [InvalidSession_CloseSession_NoErrorReported](https://github.com/ni/grpc-device/blob/0233cdf66136d6ed377282d87ce916e4d416b9f2/source/tests/system/niscope_session_tests.cpp#L115)

 So it seems like it was left out purposefully to not set status for close APIs.
 @jonlugoatx any thoughts about whether that behaviour should be retained or can I update that test to check for error after making the suggested changes?

Reply:
I have made the suggested changes. Also updated the tests which were failing. Also ran clang. @reckenro , @jonlugoatx , could you please re-review the changes as I will reset the votes?